Integrations/integration-PhishMe.yml

```diff
@@ -70,34 +94,61 @@ script:
         }
     };
 
-    var addMD = function(threats, k) {
+    var addMD = function(threat) {
         var md = "";
-        md += '### Threat ID ' + (k+1) + ': ' + threats[k].id + '\n';
+        md += '### Threat ID: ' + threat.id + '\n';
 
-        if (threats[k].label !== undefined) {
-            md += "Name of the campaign: **" + threats[0].label + "**\n";
+        if (threat.label !== undefined) {
+            md += "Name of the campaign: **" + threat.label + "**\n";
         }
-        if (threats[k].blockSet[0].ipDetail !== undefined) {
-            md += "ASN: **" + threats[k].blockSet[0].ipDetail.asn + " " + threats[k].blockSet[0].ipDetail.asnOrganization+"**\n";
+        if (threat.blockSet[0].ipDetail !== undefined) {
+            md += "ASN: **" + threat.blockSet[0].ipDetail.asn + " " + threat.blockSet[0].ipDetail.asnOrganization+"**\n";
         }
-        if (threats[k].blockSet[0].ipDetail !== undefined) {
-            md += "Country: **" + threats[k].blockSet[0].ipDetail.countryName + "**\n";
+        if (threat.blockSet[0].ipDetail !== undefined) {
+            md += "Country: **" + threat.blockSet[0].ipDetail.countryName + "**\n";
         }
-        if (threats[k].executiveSummary !== undefined) {
+        if (threat.executiveSummary !== undefined) {
             md += "#### Executive Summary: \n";
         }
-        if (threats[k].executiveSummary !== undefined) {
-            md += threats[k].executiveSummary + "\n";
+        if (threat.executiveSummary !== undefined) {
+            md += threat.executiveSummary + "\n";
         }
 
         md += "#### Threat Types:\n";
-        for (var j = 0; j < threats[k].malwareFamilySet.length; j++){
-            md += "* " + threats[k].malwareFamilySet[j].description + "\n";
+        for (var j = 0; j < threat.malwareFamilySet.length; j++){
+            md += "* " + threat.malwareFamilySet[j].description + "\n";
         }
-        md += '##### Last published: ' + new Date(threats[k].lastPublished) +'\n';
+        md += '##### Last published: ' + new Date(threat.lastPublished) +'\n';
         return md;
     };
 
+    // Returns {severityScore: 'The severity score found', md: 'The markdown for the threat', indicatorFound: 'set to true if indicator was found in threat'}
+    var checkThreat = function(threat, threshold, indicator) {
+        var severityLevel = 0;
+        var md = '';
+        var threshold_score = calcScore(threshold);
+        res = {}
+        for (var i = 0; i < threat.blockSet.length; i++){
+            if (threat.blockSet[i]['impact']) {
+                var threat_score = calcScore(threat.blockSet[i]['impact']);
+                var adjusted_score = threshold_score <= threat_score ? 3 : threat_score;
+                // if the queried indicator has a severity level, we'll take it
+                if (threat.blockSet[i]['data'] === indicator) {
```

Comment:
is it possible that we will have more than one block set with the indicator?
I think in that case we will just take the first severity level

Reply:
I haven't encountered such a scenario (but have thought of it). Taking that into account, this is exactly what will happen in such a situation - once a block set with the indicator is encountered the `indicatorFound` flag will be turned on, and the functions who call `checkThreat` will turn their flag on as well. The result is that subsequent calls to `checkThreat` will not update the dbot score (i.e. take the first value).
You're probably asking, why call `checkThreat` after encountering the indicator - the answer is to update the markdown, so that it will show the full analysis.